### DISCUSSION TITLE
mac option+number sending escape sequence instead of the respective symbol

### DISCUSSION THREAD

**Discussion by fahmij8** [sourcelair/xterm.js#5060] [Q&A] [answered]
I have xterm terminal with attach addon. When I tried the one on demo, pressing mac option+number will show the respective symbol properly
<img width="1119" alt="image" src="https://github.com/xtermjs/xterm.js/assets/58651943/f8b0a55c-37b4-478b-8520-0e34e336fe6d">

However, when I tried in my local, pressing mac option+number will show escape sequence

<img width="900" alt="image" src="https://github.com/xtermjs/xterm.js/assets/58651943/18869d42-674c-410d-bdd1-cc5fd04dacc6">

<img width="246" alt="image" src="https://github.com/xtermjs/xterm.js/assets/58651943/d7a9bd98-e6c7-4d0f-a311-5e821c1f1a6c">

the escape sequence trigger the readline arguments as seen in the above screenshot

What did I miss?

## Details
- Browser and browser version: 
Chrome: Version 125.0.6422.112
- OS version:
Mac OS: Monterey v12.6.2 
- xterm.js version: 
@xterm/xterm: ^5.4.0
custom attach addon, taken from @xterm/addon-attach 0.10.0

**Comment by fahmij8** [comment]
kindly help @Tyriar @jerch

**Comment by fahmij8** [comment]
I figured in the demo, if we activate/deactivate `macOptionIsMeta` it will switch from showing symbol to readline arguments


https://github.com/xtermjs/xterm.js/assets/58651943/b9660acd-d395-4841-951d-3fc58edadf90

**Reply by fahmij8** [reply]
However changing this on the fly still doesn't work for me


https://github.com/xtermjs/xterm.js/assets/58651943/24ce6cb1-c059-4c06-baaa-c621975ffaa5

**Reply by jerch** [reply, marked as answer]
As far as I am aware, the demo does no additional magic here, thus you should also be able to hot-toggle it. Next stops to check would be:
- different browser engine / version?
- some JS framwork interferring with keypresses?

**Reply by fahmij8** [reply]
Thank you for your reply!

For browser engine, I have checked in the latest chrome and firefox version.
and for JS Framework, I have checked it too by removing all package in one page, so the only package that I'm using is react (from next.js). Yet somehow it's still sending the escape sequence (esc+number)

**Reply by fahmij8** [reply]
I have a new finding, when I tried input `Option+number` in terminal that I created in vite. It's sending the proper symbol.
However, when I tried input `Option+number` in terminal that I created in fresh next.js project. it's sending escape sequence instead.

Can you help me confirm this @jerch?
I would like to rise this as an issue if so

**Reply by jerch** [reply]
Sorry, seems I can't help you further. I dont develop on macos and also I dont know what "terminal that I created in vite" is supposed to mean. Do you mean the JS bundler vite? How is that bundler linked to a terminal emulator?

**Reply by fahmij8** [reply]
No worries, if you want to try it out you can create a fresh vite project and setup xterm (with attach addon and ws)
same thing with next js, you can create fresh next js project and setup xterm (with attach addon and ws)

for option replacement, in windows/linux I suppose Alt should be the replacement, so you can see whether it was sending symbol or escape sequences

I'm still unsure where the problem was (FE/BE side). However I believe this is FE problem as it was being sent to the ws. At the moment I'm going to mark your previous comment as answer. Thanks for helping out!

**Reply by jerch** [reply]
Are you using SSR (server side rendering) by any chance? I think that wont work, as xterm.js is a heavy custom frontend component relying on many frontend states, which any SSR trickery most likely will get wrong. It might work with the DOM renderer to some degree, still the glyph metrics will be wrong (unless the SSR backend shim implements a full DOM engine with correct font metrics, which none does as far as I am aware).

**Reply by fahmij8** [reply]
Fortunately in next.js I use "use client" for the components xterm live
So, I'm not using any SSR
